internal/charmstore/search_test.go

```diff
@@ -804,3 +873,60 @@ func (s *StoreSearchSuite) TestUpdateConflict(c *gc.C) {
 	c.Assert(err, gc.Equals, nil)
 	c.Assert(updated, gc.Equals, false)
 }
+
+func (s *StoreSearchSuite) TestMultiSeriesCharmFiltersSeriesCorrectly(c *gc.C) {
+	charmArchive := storetesting.Charms.CharmDir("multi-series")
+	url := newResolvedURL("cs:~charmers/juju-gui-25", -1)
+	err := s.store.AddCharmWithArchive(url, charmArchive)
+	c.Assert(err, gc.IsNil)
+	err = s.store.SetPerms(&url.URL, "read", url.URL.User, params.Everyone)
+	c.Assert(err, gc.IsNil)
+	err = s.store.UpdateSearch(url)
+	c.Assert(err, gc.IsNil)
+	s.store.ES.Database.RefreshIndex(s.TestIndex)
+	filterTests := []struct {
+		series   string
+		notFound bool
+	}{{
+		series: "trusty",
+	}, {
+		series: "vivid",
+	}, {
+		series:   "sauch",
+		notFound: true,
+	}}
+	for i, test := range filterTests {
+		c.Logf("%d. %s", i, test.series)
+		res, err := s.store.Search(SearchParams{
+			Filters: map[string][]string{
+				"name":   []string{"juju-gui"},
+				"series": []string{test.series},
+			},
+		})
+		c.Assert(err, gc.IsNil)
+		if test.notFound {
+			c.Assert(res.Results, gc.HasLen, 0)
+			continue
+		}
+		c.Assert(res.Results, gc.HasLen, 1)
+		c.Assert(res.Results[0].URL.String(), gc.Equals, url.String())
+	}
+}
+
+func (s *StoreSearchSuite) TestMultiSeriesCharmSortsSeriesCorrectly(c *gc.C) {
+	charmArchive := storetesting.Charms.CharmDir("multi-series")
+	url := newResolvedURL("cs:~charmers/juju-gui-25", -1)
+	err := s.store.AddCharmWithArchive(url, charmArchive)
+	c.Assert(err, gc.IsNil)
+	err = s.store.SetPerms(&url.URL, "read", url.URL.User, params.Everyone)
+	c.Assert(err, gc.IsNil)
+	err = s.store.UpdateSearch(url)
+	c.Assert(err, gc.IsNil)
+	s.store.ES.Database.RefreshIndex(s.TestIndex)
+	var sp SearchParams
+	sp.ParseSortFields("-series")
+	res, err := s.store.Search(sp)
+	c.Assert(err, gc.IsNil)
+	c.Assert(len(res.Results) > 1, gc.Equals, true, gc.Commentf("expecting more than one result %#v", res.Results))
+	c.Assert(res.Results[0].URL.String(), gc.Equals, url.String())
```

Comment:
The sorting is tested elsewhere. This is more to test that if you have more than one series in a document it gets sorted into the correct place.